File: js/sortFunctions/shell.js
const shell = (vector, len) => {
    const start = new Date().getTime();

    let j, h, aux
    for (h = len / 2; h > 0; h /= 2) {
        for (i = h; i < len; i++) {
            aux = vector[i]
            for (j = i; j >= h && vector[j - h] > aux; j -= h)
                vector[j] = vector[j - h]
            vector[j] = aux
        }
    }
    
    const finish = new Date().getTime()
    const time = finish - start
    console.log(`Shell ${len} - time: `, time)
    if (len === 100000) {
        console.log('-----------------')
    }
}

module.exports = {
    shell
}

File: python/sortFunctions/shell.py
import datetime
import math

def shell (vector, len):
    start = datetime.datetime.now()

    h = math.floor(len / 2)
    while (h > 0):
        i = h
        while (i < len):
            aux = vector[i]
            j = i
            while (j >= h and vector[j - h] > aux):
                vector[j] = vector[j - h]
                j -= h
            vector[j] = aux
            i += 1
        h = math.floor(h / 2)
    
    finish = datetime.datetime.now()
    time = finish - start
    print(f'Shell {len} - time: {time.total_seconds() * 1000}')